.github/pull-request-template.md

```diff
@@ -1,10 +1,10 @@
-Confirm these statements before you submit your pull request:
-
-- [ ] I have read and submitted the required [Contributor Licence Agreements](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#agreements-and-licenses).
```

Comment:
I think checkbox are a better way to ensure that everyone did required points before making a PR. This is a common template for pull requests across the whole Github, so I'd keep them.

Reply:
Honestly, I don't like them, as they are not always relevant. The changes I made are also more similar to the PR template from Kubernetes.

Reply:
They are not always relevant, but they need to be read and checked. If you have a numbered list, it can be easily skipped because it doesn't require user interaction. 

It's good to see how Kubernetes does it, but I wouldn't blindly follow it in every case as there are many other good open source projects, which take different approaches. We should consider many popular solutions. Here, in PR templates, checkboxes are a better way to make sure that users read requirements and adjust PRs to them.

Reply:
> Would refrain from having checkbox formatted list in the pull request, as these do not bring much value, as being attached to every PR. Once i have accepted the CLA i do not have to tell anybody again that i have done this. This should be a label (at the moment WIP).

Yes, CLA checkbox is redundant, if we want to still have the CLA bot. But as I wrote, other checkbox need user interaction, which means he/she has to read the point to confirm it or not. What we want to do is to improve quality of PRs, and this is possible only when users read closely all requirements.

> As for the labels in the release-notes - this is fine too, however the solution proposed here imho has a purpose that a "meaningful" release-note text is written, which may not necessarily be same with the PR summary.

Then IMO the pull request is too complex and should be split - it should concentrate on one task (bring one functionality, do one/more bug fixes around one topic etc.)

> Absence of a release-note label does not mean explicitly it should not be in the release note. It means the label may not be set-up.

It would mean exactly that the PRs shouldn't be included release notes - all labels can be set up and added later if necessary (before merging PRs as well!). However, we should have just fixed number of labels.

I understand your different point of view, but I think we shouldn't bring additional complexity and some kind of redundancy to the process of making a new PR. I think if something can be simplified and automated (at least partially), it should be done so. That's my point of view.

Reply:
- Usually in Open Source projects there's a tendency to try and have as much automated/ready made bookkeeping work as possible to free up the reviewers and shorten the feedback cycle.
- IMHO the role of the checkbox is being a visual reminder/ TODO list for the creator of the pull request that's kept where it actually matters, i.e. at the Pull Request itself. Of course that doesn't mean that we should blindly trust he tested and updated the documentation because they were checked, but if you do a statistic on how many pull requests were missing tests/documentation for project that don't have these checkboxes and those that do, I am welling to bet that the latter will be way less and way improved just by adding these visual reminders.
- Anything that's automated via a bot "e.g. CLA" should be removed from the checkbox. But testing his changes, updating the documentation, following the Git workflow. Those can't and thus IMHO should stay.
- Point two "This pull request follows the contributing guidelines, recommended Git workflow, and templates" should probably be rephrased to be more concrete. "e.g. I have rebased my branch, I have good commit message etc."

Reply:
Please consider that the checkboxes are not functioning as "my task list" for PR quality properly. Anyone can check/de-check the checkbox on GitHub at any time.  So the "I" in the PR is  something we cannot rely on. 

If we want to make sure that people follow "I have tested the changes", "I have updated the documentation" etc. we can include the link to properly written guidelines in each PR, rather than a checkbox.

It also need to be the task of the PR approvers to make sure that these guidelines are followed-up. At the end, the more automation we have, including the automation of the test and the test coverage and those kind of statistics - is key - these are in fact not the statistics based on how many people claimed they did something or not. Even if it is a reminder. 

Imho, checklists serving the purpose of the PR quality and acting as reminders should not be part of the PR comment.